types/prototypes.d.ts

```diff
@@ -123,77 +107,80 @@ interface NumberConstructor {
    * @param n - A value to test
    * @returns Is it a number?
    */
-  isNumeric: (n: any) => boolean;
+  isNumeric(n: unknown): n is number;
 }
 
 /* -------------------------------------------- */
 /* Array Methods                                */
 /* -------------------------------------------- */
 
-interface ArrayConstructor {
-  fromRange: (n: number) => number[];
+declare interface ArrayConstructor {
+  fromRange(n: number): number[];
 }
 
-interface Array<T> {
-  deepFlatten: () => T[];
+type Flattened<T> = T extends Array<infer U> ? Flattened<U> : T;
```

Comment:
We should maybe put this in a namespace (for example `Array`), to not unnecessarily pollute the global namespace.

Reply:
I intentionally did not `declare` this type. I though that would prevent it from being accessible by users of the type..? Not sure though anymore, I'm going to check.

Reply:
Not sure if that's a difference. Because those prototypes were already working for me before you put the declare in front of them. But then again that could also just be a weird artifact of their global nature.

Reply:
yeah, it still makes sense to put these kinds of utility types somewhere else... I don't really know what the best place would be though. Any suggestions?

Reply:
I moved it to a new file called `utilityTypes.d.td`. I can also do it in `prototypes.d.ts, too, and put it in a namespace called `Array` if you prefer that.